test/services/database/entities/SignatureRequestsEntityService.test.ts

```diff
@@ -0,0 +1,224 @@
+import { Kysely } from "kysely";
+import { container } from "tsyringe";
+import { beforeEach, describe, expect, it, vi } from "vitest";
+import { DataKyselyService } from "../../../../src/client/kysely.js";
+import { GetSignatureRequestsArgs } from "../../../../src/graphql/schemas/args/signatureRequestArgs.js";
+import { SignatureRequestsService } from "../../../../src/services/database/entities/SignatureRequestsEntityService.js";
+import type { DataDatabase } from "../../../../src/types/kyselySupabaseData.js";
+import {
+  createTestDataDatabase,
+  generateMockAddress,
+} from "../../../utils/testUtils.js";
+
+const mockDb = vi.fn();
+
+vi.mock("../../../../src/client/kysely.js", () => ({
+  get DataKyselyService() {
+    return class MockDataKyselyService {
+      getConnection() {
+        return mockDb();
+      }
+      get db() {
+        return mockDb();
+      }
+    };
+  },
+  get kyselyData() {
+    return mockDb();
+  },
+}));
+
+// Helper function to generate mock signature request data
+function generateMockSignatureRequest() {
+  return {
+    safe_address: generateMockAddress(),
+    message_hash: `0x${Array.from({ length: 64 }, () => Math.floor(Math.random() * 16).toString(16)).join("")}`,
+    status: "pending" as const,
+    purpose: "update_user_data" as const,
+    message: JSON.stringify({ test: "data" }),
+    timestamp: Math.floor(Date.now() / 1000),
+    chain_id: 1,
+  };
+}
+
+describe("SignatureRequestsService", () => {
+  let signatureRequestsService: SignatureRequestsService;
+  let db: Kysely<DataDatabase>;
+
+  beforeEach(async () => {
+    vi.clearAllMocks();
+
+    // Create test database with signature_requests table
+    ({ db } = await createTestDataDatabase());
```

Comment:
Why is this statement wrapped in brackets?

Reply:
Because that allows to assigning the returned `db` value to the `db` defined in the test suite

Reply:
no, at least not according to my IDE